### PR TITLE
#280 - Django Auth with GEOAxIS integration

### PR DESCRIPTION
**Pull request by gisjedi**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included
- [x] documentation is changed or added

### Affected app(s)

- accounts (new)
- batch
- diagnostic
- error
- ingest
- job
- metrics
- node
- product
- queue
- recipe
- scheduler
- source
- storage

### Description of change

This PR has far reaching implications for Scale API endpoints, as all service calls now require authentication by default. There are 2 basic roles that are now available.

- Authenticated User: Able to use any safe method to perform any read-only action
- Staff User: Able to use both safe and unsafe method to perform read/write actions

The default deployment of Scale creates a single superuser account `admin`. The password is autogenerated and written to the Scale scheduler container on first launch. It is intended that it be set for a production deploy through the use of the `ADMIN_PASSWORD` environment variable. This variable may be populated using a DCOS Secret to protect the value.

The admin user can be used to bootstrap additional users by way of POST to `/accounts/users/` endpoint. Users can also be added using the Django admin UI at `/admin/` Setting the `is_staff` field to true or false determines their level of privilege.

Documentation updated for auth at: https://github.com/ngageoint/scale/wiki/Authentication-in-Scale

### CODE REVIEW

File: scale/diagnostic/views.py [hunk 1]
```diff
@@ -4,8 +4,10 @@
 
 from django.http.response import Http404
 import rest_framework.status as status
+from rest_framework.decorators import permission_classes
```
Comment (emimaesmith) [inline scale/diagnostic/views.py]: Remove unused imports `permission_classes` and `IsAdminUser`

File: scale/ingest/views.py [hunk 2]
```diff
@@ -6,7 +6,9 @@
 
 import rest_framework.status as status
 from django.http.response import Http404
+from rest_framework.decorators import permission_classes
```
Comment (emimaesmith) [inline scale/ingest/views.py]: Remove unused imports `permission_classes` and `IsAdminUser`

File: scale/job/views.py [hunk 3]
```diff
@@ -9,8 +9,10 @@
 from django.utils import timezone
 from job.seed.exceptions import InvalidSeedManifestDefinition
 from job.seed.manifest import SeedManifest
+from rest_framework.decorators import permission_classes
```
Comment (emimaesmith) [inline scale/job/views.py]: Remove unused imports `permission_classes` and `IsAdminUser`

File: scale/node/views.py [hunk 4]
```diff
@@ -5,7 +5,9 @@
 
 import rest_framework.status as status
 from django.http.response import Http404
+from rest_framework.decorators import permission_classes
```
Comment (emimaesmith) [inline scale/node/views.py]: Remove unused imports `permission_classes` and `IsAdminUser`

File: scale/queue/test/test_views.py [hunk 5]
```diff
@@ -16,9 +16,11 @@
 import util.rest as rest_util
 from job.models import Job
 from queue.models import Queue
+from rest_framework.test import APITransactionTestCase, APITestCase
+from util import rest
```
Comment (emimaesmith) [inline scale/queue/test/test_views.py]: Remove unused imports `from util import rest` and `APITestCase`

File: scale/recipe/views.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
 from django.utils.timezone import now
 
 from rest_framework.generics import GenericAPIView, ListAPIView, RetrieveAPIView, ListCreateAPIView
+from rest_framework.permissions import IsAdminUser
```
Comment (emimaesmith) [inline scale/recipe/views.py]: Remove unused import `IsAdminUser`

File: scale/scheduler/views.py [hunk 7]
```diff
@@ -9,7 +9,10 @@
 from django.http.response import Http404
 from django.utils.dateparse import parse_datetime
 from django.utils.timezone import now
+from rest_framework import permissions
+from rest_framework.decorators import permission_classes
 from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import IsAdminUser
```
Comment (emimaesmith) [inline scale/scheduler/views.py]: Unused import `IsAdminuser`?

File: scale/storage/views.py [hunk 8]
```diff
@@ -6,7 +6,9 @@
 import rest_framework.status as status
 from django.http.response import Http404, HttpResponse
 from django.utils.timezone import now
+from rest_framework.decorators import permission_classes
```
Comment (emimaesmith) [inline scale/storage/views.py]: Remove unused imports `permission_classes` and `IsAdminUser`

File: scale/accounts/management/commands/scale_superuser.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
+"""Defines the command line method for running the Scale Hello job"""
```
Comment (mikenholt) [inline scale/accounts/management/commands/scale_superuser.py]: ⌘+c / ⌘ + v remnant

File: scale/accounts/urls.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
+"""Defines the URLs for the RESTful job services"""
```
Comment (mikenholt) [inline scale/accounts/urls.py]: ⌘+c / ⌘ + v remnant

File: scale/accounts/urls.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+from __future__ import unicode_literals
+
+from django.conf.urls import url
+from django.contrib.auth import views as auth_views
```
Comment (mikenholt) [inline scale/accounts/urls.py]: Is this used?

**Pull request merged**